api/core/v2/asset.go

```diff
@@ -13,11 +12,28 @@ import (
 	"github.com/sensu/sensu-go/js"
 )
 
-// AssetNameRegexStr used to validate name of asset
-var AssetNameRegexStr = `[a-z0-9\/\_\.\-]+`
+const (
+	// AssetsResource is the name of this resource type
+	AssetsResource = "assets"
+)
+
+var (
+	// AssetNameRegexStr used to validate name of asset
+	AssetNameRegexStr = `[a-z0-9\/\_\.\-]+`
+
+	// AssetNameRegex used to validate name of asset
+	AssetNameRegex = regexp.MustCompile("^" + AssetNameRegexStr + "$")
+)
 
-// AssetNameRegex used to validate name of asset
-var AssetNameRegex = regexp.MustCompile("^" + AssetNameRegexStr + "$")
+// StorePath returns the path prefix to assets in the store
+func (a *Asset) StorePath() string {
+	return AssetsResource
+}
+
+// URIPath returns the path component of an asset URI.
+func (a *Asset) URIPath() string {
+	return path.Join(URLPrefix, "namespaces", url.PathEscape(a.Namespace), AssetsResource, url.PathEscape(a.Name))
```

Comment:
I _think_ it's unlikely to be a problem, but the old `Sprintf()` implementation would return a string with a trailing `/` if `a.Namespace` is `""`. This implementation with `path.Join()` doesn't.

Reply:
The only place in our codebase where we use the `URIPath` method (for actual logic, and not printing something) is: https://github.com/sensu/sensu-go/blob/5ffdd077e8df24d76bafa769577fa78f5cb8e3cd/cli/client/generic.go#L109

We can safely assume that we have a resource name at this place since we are doing a PUT request. So basically yeah, it won't cause any problem!